pkg/interface/src/views/apps/links/components/LinkSubmit.tsx

```diff
@@ -0,0 +1,230 @@
+  const [linkValue, setLinkValueHook] = useState('');
+  const [linkTitle, setLinkTitle] = useState('');
+  const [disabled, setDisabled] = useState(false);
+  const [linkValid, setLinkValid] = useState(false);
+
+  const doPost = () => {
+    const url = linkValue;
+    const text = linkTitle ? linkTitle : linkValue;
+    setDisabled(true);
+    const parentIndex = props.parentIndex || '';
+    const post = createPost([
+      { text },
+      { url }
+    ], parentIndex);
+
+    props.api.graph.addPost(
+      `~${props.ship}`,
+      props.name,
+      post
+    ).then(() => {
+      setDisabled(false);
+      setLinkValue('');
+      setLinkTitle('');
+      setLinkValid(false);
+    });
+  };
+
+  const validateLink = (link) => {
+    const URLparser = new RegExp(
+      /((?:([\w\d\.-]+)\:\/\/?){1}(?:(www)\.?){0,1}(((?:[\w\d-]+\.)*)([\w\d-]+\.[\w\d]+))){1}(?:\:(\d+)){0,1}((\/(?:(?:[^\/\s\?]+\/)*))(?:([^\?\/\s#]+?(?:.[^\?\s]+){0,1}){0,1}(?:\?([^\s#]+)){0,1})){0,1}(?:#([^#\s]+)){0,1}/
+    );
+
+    let linkValid = URLparser.test(link);
+
+    if (!linkValid) {
+      linkValid = URLparser.test(`http://${link}`);
+      if (linkValid) {
+        link = `http://${link}`;
+        setLinkValue(link);
+      }
+    }
+
+    if (linkValid) {
+      if (hasProvider(linkValue)) {
+        fetch(`https://noembed.com/embed?url=${linkValue}`)
+          .then(response => response.json())
+          .then((result) => {
+            if (result.title && !linkTitle) {
+              setLinkTitle(result.title);
+            }
+          }).catch((error) => { /* noop*/ });
+      } else if (!linkTitle) {
+        setLinkTitle(decodeURIComponent(link
+          .split('/')
+          .pop()
+          .split('.')
+          .slice(0, -1)
+          .join('.')
+          .replace('_', ' ')
+          .replace(/\d{4}\.\d{1,2}\.\d{2}\.\.\d{2}\.\d{2}\.\d{2}-/, '')
+        ));
+      }
+    }
+    return link;
+  };
+
+  const onFileDrag = useCallback(
+    async (files: FileList | File[], e: DragEvent) => {
+      if (!canUpload) {
+        return;
+      }
+      uploadDefault(files[0]).then(setLinkValue);
+    },
+    [uploadDefault, canUpload]
+  );
+
+  const { bind, dragging } = useFileDrag(onFileDrag);
+
+  const onLinkChange = (linkValue: string) => {
+    setLinkValueHook(linkValue);
+    const link = validateLink(linkValue)
+    setLinkValid(link);
+  };
+
+  const setLinkValue = (linkValue: string) => {
+    onLinkChange(linkValue);
+    setLinkValueHook(linkValue);
+  };
+
+  const onPaste = useCallback(
+    (event: ClipboardEvent) => {
+      if (!event.clipboardData || !event.clipboardData.files.length) {
+        return;
+      }
+      event.preventDefault();
+      event.stopPropagation();
+      uploadDefault(event.clipboardData.files[0]).then(setLinkValue);
+    }, [setLinkValue]
+  );
+
+  const onKeyPress = useCallback(
+    (e) => {
+      if (e.key === 'Enter') {
+        e.preventDefault();
+        doPost();
+      }
+    }, []
```

Comment:
doPost missing from dependencies, but because it's not memoised you can probably drop the useCallback here entirely